src/game.cpp

```diff
@@ -5487,12 +5477,23 @@ void Game::parsePlayerExtendedOpcode(uint32_t playerId, uint8_t opcode, const st
 	}
 }
 
-std::forward_list<Item*> Game::getMarketItemList(uint16_t wareId, uint16_t sufficientCount, DepotChest* depotChest, Inbox* inbox)
+std::forward_list<Item*> Game::getMarketItemList(uint16_t wareId, uint16_t sufficientCount, Player* player)
 {
 	std::forward_list<Item*> itemList;
```

Comment:
Please make it `std::vector` instead. Iterating over lists is terrible, and if we are trying to avoid allocation cost (I think it's insignificant), just `reserve(sufficientCount)` and call it a day.